api/src/main/java/bio/terra/tanagra/indexing/BigQueryIndexingJob.java

```diff
@@ -51,17 +81,37 @@ protected JobStatus checkTableExistenceForJobStatus(TablePointer outputTable) {
 
   protected void createTableFromSql(TablePointer outputTable, String sql, boolean isDryRun) {
     BigQueryDataset outputBQDataset = getOutputDataPointer();
-    LOGGER.info(
-        "output BQ table: project={}, dataset={}, table={}",
-        outputBQDataset.getProjectId(),
-        outputBQDataset.getDatasetId(),
-        outputTable.getTableName());
-
     TableId destinationTable =
         TableId.of(
             outputBQDataset.getProjectId(),
             outputBQDataset.getDatasetId(),
             outputTable.getTableName());
     outputBQDataset.getBigQueryService().createTableFromQuery(destinationTable, sql, isDryRun);
   }
+
+  protected BigQueryOptions buildDataflowPipelineOptions(BigQueryDataset outputBQDataset) {
+    // TODO: Allow overriding the default region.
+    String[] args = {
+      "--runner=dataflow",
+      "--project=" + outputBQDataset.getProjectId(),
+      "--region=" + DEFAULT_REGION,
+      "--serviceAccount=" + getAppDefaultSAEmail(),
+    };
+    // TODO: Use PipelineOptionsFactory.create() instead of fromArgs().
+    // PipelineOptionsFactory.create() doesn't seem to call the default factory classes, while
+    // fromArgs() does.
+    return PipelineOptionsFactory.fromArgs(args).withValidation().as(BigQueryOptions.class);
+  }
+
+  protected String getAppDefaultSAEmail() {
```

Comment:
Would this method be used by other pipelines? Maybe extract it to its own service if that's the case. I don't feel it belongs in this class.

Reply:
All existing pipelines extend this base class. Specifically, it's needed to build the Dataflow options object for kicking off a Pipeline. We expect to add new indexing jobs later that e.g. run on Azure instead of BQ/Dataflow, and those would likely need a new base class. I'm not sure whether those would also need to pull the app default SA email, but if they do, then I think it would make sense to pull this out into a shared class at that point.

Reply:
should be fine for now I think, just noticed it's in the abstract BigQueryIndexingJob and not in the specific implementation